File: 40130112106/maximumSubarraySum.js
function maximumSubarraySum(array) {
  let n = array.length;
  let max = array[0];
  for (let i = 0; i < n; i++) {
    let s = array[i];
    if (s >= max) {
      max = s;
    }
    let j = i + 1;
    while (j < n) {
      s += array[j];
      if (s >= max) {
        max = s;
      }
      j++;
    }
  }
  return max;
}

let array = [1, 5, -1, -50, 3, 4, 7, -5];
console.log(maximumSubarraySum(array));

File: 40130112056/Maximum Subarray Sum.py
def MaximumSubarraySum(Array):
    n=len(Array)
    Max=Array[0]
    for i in range(0,n):
        s=Array[i]
        if ( s >= Max ):
            Max=s
        j=i+1
        while(j < n):
            s = s + Array[j]
            if ( s >= Max ):
                Max=s
            j=j+1

    return Max

# array = [-2, -1, -3, 4, -1, 2, 1, -5, 4]
array=[1 ,5 ,-1 , -50 , 3 , 4 ,7 , -5]
print(MaximumSubarraySum(array))
